website/content/deployment/example-firebase.mdx

```diff
@@ -0,0 +1,202 @@
+With _deploy targets_ you can uniquely identify a specific site by its _target name_ in your `firebase.json` configuration file and in your Firebase CLI commands for testing or deploying to your sites.
+
+To create a deploy target and apply a target name to a Hosting site, run the following CLI command from the root of your project directory:
+
+```
+$ firebase target:apply hosting target-name resource-name
+```
+
+Where the parameters are:
+
+- **target-name** — a unique identifier (that you've defined yourself) for the Hosting site that you're deploying to
+
+- **resource-name** — the name of the Hosting site as listed in your Firebase project
+
+![Firebase Hosting Console](/assets/custom-applications/mc-dev-firebase-hosting-console.png)
+
+The settings for deploy targets are stored in the `.firebaserc` file in your project directory, so you only need to set up deploy targets one time per project.
+
+## Configuration
+
+First, you need to create an `env.prod.json` file with the following JSON:
+
+```json
+{
+  "applicationName": "my-app",
+  "frontendHost": "mc.commercetools.co",
+  "mcApiUrl": "https://mc-api.commercetools.com",
+  "location": "eu",
+  "env": "production",
+  "cdnUrl": "https://[projectID].firebaseapp.com",
+  "servedByProxy": true
+}
+```
+
+You also need a `csp.json` file. You can also have a `csp.prod.json` file if you want to use a different configuration for `development`.
+
+```json
+{
+  "script-src": ["[projectID].firebaseapp.com"],
+  "connect-src": ["[projectID].firebaseapp.com"],
+  "style-src": ["[projectID].firebaseapp.com"]
+}
+```
+
+For both `env.prod.json` and `csp.json`, `[projectID]` should be replaced with your Firebase project ID.
+
+## Build and Compilation
+
+In order to generate the correct configuration for Firebase, the `mc-scripts compile-html` script must use a Firebase transformer. This transformer will generate a `firebase.json` file with the custom security headers required for Merchant Center Custom Applications.
+
+```javascript
+const fs = require('fs');
+const path = require('path');
+
+module.exports = ({ headers }) => {
+  const baseConfig = {
+    hosting: {
+      public: 'public',
+      ignore: ['firebase.json', '**/.*', '**/node_modules/**'],
+      rewrites: [
+        {
+          source: '**',
+          destination: '/index.html',
+        },
+      ],
+      headers: [
+        {
+          source: '**/*.@(js.map|js|css|txt|html|png)',
+          headers: [
+            {
+              key: 'Cache-Control',
+              value: 's-maxage=31536000,immutable',
+            },
+          ],
+        },
+        {
+          source: '**',
+          headers: Object.entries({
+            ...headers,
+            'Cache-Control': 'no-cache',
+          }).map(([key, value]) => ({
+            key,
+            value,
+          })),
+        },
```

Comment:
Suggestion: we can maybe define a "fallback" route for the login/logout pages as I did in the playground example for now: https://github.com/commercetools/merchant-center-application-kit/blob/master/playground/config/fallback-route.js

This should prevent the infinite redirect loop and provide some helpful information. WDYT?

Reply:
I will look into it! Not sure if Firebase (or AWS) support this feature.